nixos/modules/security/apparmor-suid.nix

```diff
@@ -21,29 +21,39 @@ with lib;
   };
 
   config = mkIf (cfg.confineSUIDApplications) {
-    security.apparmor.profiles = [ (pkgs.writeText "ping" ''
-      #include <tunables/global>
-      /run/wrappers/bin/ping {
-        #include <abstractions/base>
-        #include <abstractions/consoles>
-        #include <abstractions/nameservice>
+    security.apparmor.policies."bin/ping".profile = ''
+      include <tunables/global>
+      /run/wrappers/wrappers.*/ping {
+        include <abstractions/base>
+        include <abstractions/consoles>
+        include <abstractions/nameservice>
 
         capability net_raw,
         capability setuid,
         network inet raw,
 
-        ${pkgs.stdenv.cc.libc.out}/lib/*.so mr,
-        ${pkgs.libcap.lib}/lib/libcap.so* mr,
-        ${pkgs.attr.out}/lib/libattr.so* mr,
-
+        ${getLib pkgs.stdenv.cc.cc}/lib/*.so* mr,
+        ${getLib pkgs.stdenv.cc.libc}/lib/*.so* mr,
+        ${getLib pkgs.stdenv.cc.libc}/lib/gconv/gconv-modules r,
+        ${getLib pkgs.glibcLocales}/lib/locale/locale-archive r,
+        ${getLib pkgs.attr.out}/lib/libattr.so* mr,
+        ${getLib pkgs.libcap.lib}/lib/libcap.so* mr,
+        ${getLib pkgs.libcap_ng}/lib/libcap-ng.so* mr,
+        ${getLib pkgs.libidn2}/lib/libidn2.so* mr,
+        ${getLib pkgs.libunistring}/lib/libunistring.so* mr,
+        ${getLib pkgs.nettle}/lib/libnettle.so* mr,
```

Comment:
how about generating this in nix with the closure of pkgs.ping?

Reply:
Could be moved to a .apparmor output even, but might be not necessary.